File: lib/api-response.ts
// lib/api-response.ts — Contrato de respuesta (Art. IX + IX v1.3)
// Toda Server Action y API Route retorna este tipo, sin excepción.

export type ApiResponse<T> =
  | { success: true; data: T; warning?: string }
  | { success: false; error: string };

/**
 * Respuesta exitosa. Opcionalmente incluye un warning que NO bloquea la operación.
 * Ejemplo de warning: "Ya existe un ticket abierto para este equipo"
 */
export function ok<T>(data: T, warning?: string): ApiResponse<T> {
  return warning
    ? { success: true, data, warning }
    : { success: true, data };
}

/**
 * Respuesta de error. Mensaje seguro para el cliente (Art. X).
 * Nunca incluye stack traces, códigos de Prisma/MySQL, ni rutas del servidor.
 */
export function fail(error: string): ApiResponse<never> {
  return { success: false, error };
}

File: EpisLab.Test/ClassLibrary1/Services/ApiResponse.cs
// Services/ApiResponse.cs
// Equivalente a: lib/api-response.ts
// Contrato de respuesta (Art. IX): { success, data?, error?, warning? }
namespace SistemaGestionLaboratorios.Services
{
    public class ApiResponse<T>
    {
        public bool Success { get; set; }
        public T? Data { get; set; }
        public string? Error { get; set; }
        public string? Warning { get; set; }
        public bool HasWarning => Warning != null;
        public bool HasData => Data != null;
    }

    public static class ApiResponseHelper
    {
        /// <summary>Respuesta exitosa. Warning opcional, no bloquea (Art. IX v1.3).</summary>
        public static ApiResponse<T> Ok<T>(T data, string? warning = null)
        {
            return new ApiResponse<T>
            {
                Success = true,
                Data = data,
                Warning = warning
            };
        }

        /// <summary>Respuesta de error. Nunca expone detalles internos (Art. X).</summary>
        public static ApiResponse<T> Fail<T>(string error)
        {
            return new ApiResponse<T>
            {
                Success = false,
                Error = error
            };
        }

        public static ApiResponse<object> Fail(string error)
            => Fail<object>(error);
    }
}
